server/src/main/java/org/elasticsearch/search/lookup/SearchLookup.java

```diff
@@ -24,20 +24,60 @@
 import org.elasticsearch.index.mapper.MappedFieldType;
 import org.elasticsearch.index.mapper.MapperService;
 
+import java.util.LinkedHashSet;
+import java.util.Set;
+import java.util.function.BiFunction;
 import java.util.function.Function;
+import java.util.function.Supplier;
 
 public class SearchLookup {
 
-    final DocLookup docMap;
+    private static final int MAX_FIELD_CHAIN = 32;
```

Comment:
32 feels big to me ;). I think we can be more aggressive here, something like 4-5 should be more than enough ?

Reply:
fine with me, I totally expected some discussion on this limit. maybe 4-5 is a bit on the low side, shall we do 10? :)

Reply:
I am maybe be too cautious here but I cannot think of a good example of a chain that would require more than 2-3 indirections. Allowing for more looks like an anti-pattern to me.

Reply:
being cautious is not a bad idea, let's start with 5 then and see if users complain.